### DISCUSSION TITLE
Resizable text boxes and text editing features

### DISCUSSION THREAD

**Discussion by starlord-brazil** [GraphiteEditor/Graphite#1963] [General]
Hello Graphite Community,

I'm exploring the capabilities of Graphite and am particularly interested in enhancing the text editing features within the software. I'm looking for a way to implement resizable text boxes where the text adjusts automatically to fit the size of the box. Additionally, I'd like the ability to configure line spacing within these text boxes. Could you provide insights or guidance on how to develop these features, or is there already work underway to integrate such functionalities into Graphite? Any pointers or collaboration on this would be greatly appreciated!

Thank you!

**Comment by 0HyperCube** [comment]
Hello @starlord-brazil, I'm delighted to read that you are interested in contributing!

The text tool is currently just a very simple prototype. We have an issue tracking some improvements we would like to make for it in #1105; however this is not currently being worked on. I assume your first point referred to automatically line wrapping based on a maximum width set by the user (the first bullet point). I've added configuring line spacing to that issue as well; thanks for reporting that.

Feel free to start with any part of that issue that interests you. The code for converting the text to a path is on `node-graph/gcore/src/text/to_path.rs:83`. The code for the text tool (which currently just uses a HTML input element whilst the text is being edited) is at `editor/src/messages/tool/tool_messages/text_tool.rs`.

If you need help, please ask on the Graphite Discord (https://discord.graphite.rs) or on the relevant issue/PR. Discord will probably get you a faster response.

If you haven't already, please review the contributing guide (https://graphite.rs/volunteer/guide/getting-started/). Thanks again for your interest.
